### ISSUE TITLE
Loading transformation rules from JSON file : Java Mapping Master

### ISSUE THREAD

**Issue by shashi792** [protegeproject/mapping-master#28]
I am trying to use the provided MMExample to use the mapping master API. But, from the code, I see the part of the code which defines the transformation rule

#25 

```
// Create a Mapping Master expression. A Mapping Master expression is rendered over a range of cells in a sheet.
      // Here we create individuals from cell range A1:A3 in a spreadsheet typed by the class Car defined in the source ontology.
      final String sheetName = "MySheet";
      final Integer startColumnNumber = 1, finishColumnNumber = 1, startRowNumber = 1, finishRowNumber = 3;
      TransformationRule mmExpression = new TransformationRule(sheetName, columnNumber2Name(startColumnNumber),
        columnNumber2Name(finishColumnNumber), startRowNumber.toString(), finishRowNumber.toString(),
        "Create car instances from column A", "Individual: @A* Types: Car");

      // Create a Mapping Master parser for the expression, parse it, and return an AST node representing the expression
      MappingMasterParser parser = new MappingMasterParser(
        new ByteArrayInputStream(mmExpression.getRuleString().getBytes()), new ReferenceSettings(), -1);
      MMExpressionNode mmExpressionNode = new ExpressionNode((ASTExpression)parser.expression()).getMMExpressionNode();
```
 
How do we ingest a json (attached) to load the transformation rules like we do it in the protege editor ?. This is what I have tired up-until now. 
```
File owlFile = new File(defaultOntology);
		File spreadSheet = new File(spreadSheetFileName);
		
		
		// Create an OWL ontology using OWL API
		OWLOntologyManager ontologyManager = OWLManager.createOWLOntologyManager();
		OWLOntology ontology = ontologyManager.loadOntologyFromOntologyDocument(owlFile);
		
		// Create workbook using Apache POI
		Workbook workbook = WorkbookFactory.create(spreadSheet);
		
		//Create ontology source and spreadsheet data source
		
		OWLOntologySource ontologySource = new OWLAPIOntology(ontology);
		SpreadSheetDataSource spreadsheetSource = new SpreadSheetDataSource(workbook);
		
		//Create Mapping Master expression
		TransformationRuleSet transformationRuleSet = TransformationRuleSetFactory.loadTransformationRulesFromDocument(ruleFileLocation);
	    Set<TransformationRule> ruleSet = transformationRuleSet.getTransformationRules();
	    
	    // Create an OWL renderer and supply it with an ontology and a spreadsheet. 
	    // An OWL renderer renders a set of OWLAPI-based OWL axioms from a Mapping Master expression.
	      OWLRenderer owlRenderer = new OWLRenderer(ontologySource, spreadsheetSource); 
	    
	      int count = 1;
	    for (TransformationRule rule : ruleSet) {
	    	 System.out.println("Rule " + count + "/" + ruleSet.size() );
	    	
	    	//Mapping Master Parser for the expression, parse it and return an AST node representing the expression
	    	MappingMasterParser parser = new MappingMasterParser(
	    	        new ByteArrayInputStream(rule.getRuleString().getBytes()), new ReferenceSettings(), -1);
	    	MMExpressionNode mmExpressionNode = new ExpressionNode((ASTExpression)parser.expression()).getMMExpressionNode();
	    	
	    	
// Loop through the cells specified by the Mapping Master expression
            for (int columnNumber = SpreadSheetUtil.columnName2Number(rule.getStartColumn()); columnNumber <= SpreadSheetUtil.columnName2Number(rule.getEndColumn()); columnNumber++) {
    	      for (int rowNumber = SpreadSheetUtil.rowLabel2Number(rule.getStartRow()); rowNumber <= SpreadSheetUtil.rowLabel2Number(rule.getEndRow()); rowNumber++) {
  
	            // A Mapping Master expression is rendered in the context of a location in a spreadsheet
	        	  spreadsheetSource.setCurrentLocation(new SpreadsheetLocation(rule.getSheetName(), columnNumber, rowNumber));
	        	  
	        	  
	        	// Render the Mapping Master expression as an OWL rendering (which will contain a set of OWLAPI-based OWL axioms)
	              Optional<OWLRendering> owlRendering = owlRenderer.render(mmExpressionNode);

	              // Display the OWL axioms rendered by the Mapping Master expression and add them to the source ontology
	              if (owlRendering.isPresent()) {
		                Set<OWLAxiom> renderedOWLAxioms = owlRendering.get().getOWLAxioms();
		                System.out.println("Rendered OWL axioms: " + renderedOWLAxioms);
		                ontologyManager.addAxioms(ontology, renderedOWLAxioms);
	              }
	          }
	        }
	        count++;
	        ontology.saveOntology();
	    }
	}
```

I get an error 

Exception in thread "main" java.lang.NoSuchMethodError: 'boolean org.semanticweb.owlapi.model.OWLDocumentFormat.isPrefixOWLOntologyFormat()'
	at org.mm.core.OWLAPIOntology.buildPrefixManager(OWLAPIOntology.java:35)
	at org.mm.core.OWLAPIOntology.getEntityResolver(OWLAPIOntology.java:28)
	at org.mm.renderer.owlapi.OWLObjectFactory.newInstance(OWLObjectFactory.java:90)
	at org.mm.renderer.owlapi.OWLRenderer.<init>(OWLRenderer.java:81)
	at com.ey.ailabs.bangalore.OntologyMappingFromSpreadSheet.mapDataFromSpreadSheet2Ontology(OntologyMappingFromSpreadSheet.java:67)

i.e.     `OWLRenderer owlRenderer = new OWLRenderer(ontologySource, spreadsheetSource); 
`

**Comment by shashi792** [comment]
This issue was resolved by downgrading the owl-api version to 4.5.10 from 5.1.17.

**Comment by shashi792** [comment]
Error while looping through row number and col numbers. 
```
for (int columnNumber = Integer.parseInt(rule.getStartColumn()); columnNumber <= Integer.parseInt(rule.getEndColumn()); columnNumber++) {
	          for (int rowNumber = Integer.parseInt(rule.getStartRow()); rowNumber <= Integer.parseInt(rule.getEndRow()); rowNumber++) {
```
With the JSON 

```
{
  "Collections": [
    {
      "sheetName": "Transactional data",
      "startColumn": "K",
      "endColumn": "L",
      "startRow": "1",
      "endRow": "+",
      "comment": "SubClass of definitions",
      "rule": "Class: @L1(mm:snakeCaseEncode) \nSubClassOf: @K1(mm:snakeCaseEncode)",
      "active": true
    },
    {
      "sheetName": "Transactional data",
      "startColumn": "A",
      "endColumn": "O",
      "startRow": "2",
      "endRow": "+",
      "comment": "",
      "rule": "Individual:@A*\n Annotations: rdfs:label @B*\n Types:@A1\n Facts: hasValue @O*(xsd:float),\n           hasVersionCategory @E*(xsd:string)",
      "active": true
    }	           
```
**rule.getStartColumn()** and similar _Integers_ will bring back **A, +** and other values which makes sense, but if so, how do we loop through then

**Comment by shashi792** [comment]
@martinjoconnor I believe you wrote that MMExample.java. Any pointers to the problem.

**Comment by shashi792** [comment]
This issue was resolved by downgrading the owl-api version to 4.5.10 from 5.1.17.
